src/Compatibility/Microsoft.DotNet.ApiCompatibility/Rules/AssemblyIdentityMustMatch.cs

```diff
@@ -0,0 +1,106 @@
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+
+namespace Microsoft.DotNet.ApiCompatibility.Rules
+{
+    public class AssemblyIdentityMustMatch : Rule
+    {
+        public override void Initialize(RuleRunnerContext context)
+        {
+            context.RegisterOnAssemblySymbolAction(RunOnAssemblySymbol);
+        }
+
+        private void RunOnAssemblySymbol(IAssemblySymbol left, IAssemblySymbol right, string leftName, string rightName, IList<CompatDifference> differences)
+        {
+            if (left == null && right != null)
+            {
+                differences.Add(new CompatDifference(DiagnosticIds.MatchingAssemblyDoesNotExist, string.Format(Resources.AssemblyNameDoesNotExist, leftName, right.Identity.Name), DifferenceType.Removed, right.Identity.GetDisplayName()));
+                return;
+            }
+
+            if (left != null && right == null)
+            {
+                differences.Add(new CompatDifference(DiagnosticIds.MatchingAssemblyDoesNotExist, string.Format(Resources.AssemblyNameDoesNotExist, rightName, left.Identity.Name), DifferenceType.Added, left.Identity.GetDisplayName()));
+                return;
+            }
+
+            AssemblyIdentity leftIdentity = left.Identity;
+            AssemblyIdentity rightIdentity = right.Identity;
+
+            string leftAssemblyName = leftIdentity.Name;
+            string leftAssemblyCulture = string.IsNullOrEmpty(leftIdentity.CultureName) ? "neutral" : left.Identity.CultureName;
+            Version leftAssemblyVersion = leftIdentity.Version;
+            ReadOnlySpan<byte> leftAssemblyPublicKeyToken = leftIdentity.PublicKeyToken.AsSpan();
+
+            string rightAssemblyName = rightIdentity.Name;
+            string rightAssemblyCulture = string.IsNullOrEmpty(rightIdentity.CultureName) ? "neutral" : right.Identity.CultureName;
+            Version rightAssemblyVersion = rightIdentity.Version;
+            ReadOnlySpan<byte> rightAssemblyPublicKeyToken = rightIdentity.PublicKeyToken.AsSpan();
+
+            if (leftAssemblyName != rightAssemblyName)
+            {
+                differences.Add(CreateDifference(Resources.AssemblyNameDoesNotMatch, leftAssemblyName, rightAssemblyName, leftName, rightName, rightIdentity));
+            }
+
+            if (leftAssemblyCulture != rightAssemblyCulture)
+            {
+                differences.Add(CreateDifference(Resources.AssembyCultureDoesNotMatch, leftAssemblyCulture, rightAssemblyCulture, leftName, rightName, rightIdentity));
+            }
+ 
+            if (rightAssemblyVersion < leftAssemblyVersion)
+            {
+                differences.Add(CreateDifference(Resources.AssembyVersionDoesNotMatch, rightAssemblyVersion.ToString(), leftAssemblyVersion.ToString(), rightName, leftName, rightIdentity));
```

Comment:
Maybe add another resource string so that the error message eludes to the version needing to be greater or equal?